packages/api/endpoints/pdrs.js

```diff
@@ -63,25 +65,27 @@ async function del(req, res) {
 
   const pdrS3Key = `${process.env.stackName}/pdrs/${pdrName}`;
 
-  await deleteS3Object(process.env.system_bucket, pdrS3Key);
-
   const pdrModel = new models.Pdr();
+  const pdrPgModel = new PdrPgModel();
+  const knex = await getKnexClient();
 
-  try {
+  await knex.transaction(async (trx) => {
+    await pdrPgModel.delete(trx, { name: pdrName });
     await pdrModel.delete({ pdrName });
+    await deleteS3Object(process.env.system_bucket, pdrS3Key);
+  }).catch((error) => {
+    res.boom.badRequest(`Error deleting PDR ${pdrName}`, error);
+    throw error;
+  });
 
-    if (inTestMode()) {
-      const esClient = await Search.es(process.env.ES_HOST);
-      await esClient.delete({
-        id: pdrName,
-        index: process.env.ES_INDEX,
-        type: 'pdr',
-      }, { ignore: [404] });
-    }
-  } catch (error) {
-    if (!isRecordDoesNotExistError(error)) throw error;
```

Comment:
So this error trap was ensuring that if the error thrown was "RecordDoesNotExist", then the endpoint would return 200. Basically DELETE returns 200 if the record is already deleted. Is that still true?

Reply:
Rolled back that part of the change.